configure.ac

```diff
@@ -4872,58 +4872,8 @@ AC_SUBST([MPI_F77_COUNT])
 
 # ----------------------------------------------------------------------------
 
-#
-# Set size of MPI_Status.  Must come after MPI_Count determination.
-#
-
-#
-# The size of MPI_Status is needed for the Fortran interface.
-#
-# WARNING!!! this is a spot where we duplicate code from mpi.h.in and it *must*
-# be kept in sync in order to make a proper computation
-AC_CACHE_CHECK([for size of MPI_Status],[pac_cv_sizeof_mpi_status],[
-dnl "double-quote" (in m4), otherwise you end up with a heinous bug because of
-dnl the array subscripting below
-[
-    rm -f pac_mpi_status.h
-    cat > pac_mpi_status.h <<_EOF
-typedef struct {
-    int count_lo;
-    int count_hi_and_cancelled;
-    int MPI_SOURCE;
-    int MPI_TAG;
-    int MPI_ERROR;
-} MPI_Status;
-_EOF
-]
-
-dnl just compute it, since a 1s-complement or sign-and-magnitude machine is
-dnl *highly* unlikely.  Users will report the error if it is ever
-dnl encountered, which will be safer than attempting some never-tested
-dnl default fallback.
-    AC_COMPUTE_INT([pac_cv_sizeof_mpi_status],
-                   [sizeof(MPI_Status)],
-                   [#include "pac_mpi_status.h"],
-                   [AC_MSG_ERROR([unable to compute status size, are you compiling on a non-2s-complement host?])])
-    rm -f pac_mpi_status.h
-])
-
-SIZEOF_MPI_STATUS=$pac_cv_sizeof_mpi_status
-export SIZEOF_MPI_STATUS
-AC_SUBST([SIZEOF_MPI_STATUS])
-
-if test "$enable_f77" = yes -a -z "$MPI_STATUS_SIZE" ; then
-    if test -n "$SIZEOF_MPI_STATUS" ; then
-        # compute from the C sizeof
-        AS_VAR_ARITH([MPI_STATUS_SIZE],[$SIZEOF_MPI_STATUS / $ac_cv_sizeof_int])
-        if test "$MPI_STATUS_SIZE" = "0" ; then
-            AC_MSG_ERROR([Could not compute the size of MPI_Status])
-        fi
-    else
-        AC_MSG_ERROR([MPI_STATUS_SIZE was not defined!])
-    fi
-fi # enable_f77 and mpi_status_size not set
-AC_SUBST([MPI_STATUS_SIZE])
+# MPI_STATUS_SIZE is hardcoded and need to be synchronized with mpi.h
+MPI_STATUS_SIZE=5
```

Comment:
We dropped the `MPI_Fint` usage from this PR, so I think this commit can also be dropped?

Reply:
Sure. But the commit is not related. In either case we'll need the MPI_STATUS_SIZE to agree with the number of integer fields in the status struct or the code else where will break. Currently we are hard coding the actual `struct` definition into the `conftest` then get the size and divide by the size of `int` to just get the number `5`, it just seems silly to me. `MPI_STATUS_SIZE` need to be `5` even in the case `MPI_Fint` and `int` have different sizes.

But yes, the commit is not essential to this PR.